src/utils/permissions.ts

```diff
@@ -0,0 +1,68 @@
+import axios from "axios";
+import memoize from "nano-memoize";
+
+import { D2 } from "../types/d2";
+
+export enum AppRoles {
+    METADATA_SYNC_ADMINISTRATOR = "METADATA_SYNC_ADMINISTRATOR",
+}
+
+export const isUserAdmin = memoize(async (d2: D2) => {
+    const userRoles = await d2.currentUser.getUserRoles();
+    return userRoles
+        .toArray()
+        .find((role: any) => role.name === AppRoles.METADATA_SYNC_ADMINISTRATOR);
+});
+
+export const getUserInfo = memoize(
+    async (
+        d2: D2
+    ): Promise<{
+        userGroups: any[];
```

Comment:
As you hinted in the PR comment, those `any` should be real interfaces when `d2-api` gets integrated.